From 93ef9849ef6367b07e8d2a0fea35b8b1d9431054 Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Sun, 9 Mar 2014 19:37:56 -0700
Subject: ASoC: simple-card: card name can be option

snd_card.name is now option on DT case.
non-DT case can follow same style,
and it is understandable from platform point of view.

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Signed-off-by: Mark Brown <broonie@linaro.org>
(cherry picked from commit 12ffa6fc1958879a81b6af1624c1a2edd83ec04a)
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
---
 sound/soc/generic/simple-card.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sound/soc/generic/simple-card.c b/sound/soc/generic/simple-card.c
index 7cabcc5c8703..5dd47691ba41 100644
--- a/sound/soc/generic/simple-card.c
+++ b/sound/soc/generic/simple-card.c
@@ -287,7 +287,6 @@ static int asoc_simple_card_probe(struct platform_device *pdev)
 		}
 
 		if (!cinfo->name	||
-		    !cinfo->card	||
 		    !cinfo->codec_dai.name	||
 		    !cinfo->codec	||
 		    !cinfo->platform	||
@@ -296,7 +295,7 @@ static int asoc_simple_card_probe(struct platform_device *pdev)
 			return -EINVAL;
 		}
 
-		priv->snd_card.name	= cinfo->card;
+		priv->snd_card.name	= (cinfo->card) ? cinfo->card : cinfo->name;
 		dai_link->name		= cinfo->name;
 		dai_link->stream_name	= cinfo->name;
 		dai_link->platform_name	= cinfo->platform;
-- 
2.1.2

